### ISSUE TITLE
Azure ID / secrets

### ISSUE THREAD

**Issue by adepierre** [maxsupermanhd/go-mc-ms-auth#2]
(not really an issue, rather a question, but discussions are not activated in this repo)

So in the ``GetMCcredentials`` you need a ``cid`` string parameter. If I understand correctly, this sould be the azure client ID as stated on [the wiki](https://wiki.vg/Microsoft_Authentication_Scheme). If no ``cid`` is passed to the function, your code tries to check in the OS env variables to find one. So my first question is: what should we pass here as ``cid``? I guess we could use the same ID used for minecraft, but I'm not sure this is allowed.

And my second question is about the secret. Still according to the wiki, a client secret is required for refreshing the initial token. At [L40](https://github.com/maxsupermanhd/go-mc-ms-auth/blob/9b45559ec8fb9bf7f7fc526b2c14ba93d150febf/auth.go#L40),  it seems that you commented out this part. Could you elaborate on that? Is the secret required or is the wiki wrong on this point?

Thanks for the code and the time you take for this 😀

**Comment by adepierre** [comment]
@maxsupermanhd Do you have any information about that?

**Comment by maxsupermanhd** [comment]
Damn, did not seen it at all, sorry about that...

TLDR register a microsoft application and setup oauth thing, I followed [this](https://docs.microsoft.com/en-us/onedrive/developer/rest-api/getting-started/app-registration?view=odsp-graph-online).

You can also use whatever clientid you want (I named it cid because was writing this only as proof of concept and for my own personal projects) or even plug in one that I made: `88650e7e-efee-4857-b9a9-cf580a00ef43` I believe that it is made just for app identification and further integration with other ms services.

As I tested no client secret is actually required - it works without any specified since it is not application level task, it is purely client sided thing so no client secret must be distributed.

Also whole auth flow is described [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code) and you can also try do it yourself.

And again, sorry for very late answer I completely missed issue and your ping in general, sometimes they don't show up.

**Comment by adepierre** [comment]
Alright, thank you for the feedback. I think I understand everything correctly now. I'm kind of busy now, but hopefully I'll be able to spare some time to work on that next week.
